src/net/http/headers.rs

```diff
@@ -0,0 +1,50 @@
+use std::collections::HashMap;
+
+#[derive(Default, Debug, Clone)]
+pub struct Headers {
+    headers: HashMap<String, String>,
+}
+
+impl Headers {
+    pub fn new() -> Headers {
+        Headers {
+            headers: HashMap::new(),
+        }
+    }
+
+    pub fn set(&mut self, key: &str, value: &str) {
+        self.headers.insert(key.to_string(), value.to_string());
+    }
+
+    pub fn get(&self, key: &str) -> Option<&String> {
+        self.headers.get(key)
+    }
+
+    /// Returns all the header entries. Note that there is no ordering in here!
+    pub fn all(&self) -> &HashMap<String, String> {
+        &self.headers
+    }
+
+    pub fn sorted(&self) -> Vec<(&String, &String)> {
+        let mut sorted = self.headers.iter().collect::<Vec<_>>();
```

Comment:
Hashmaps are not deterministic in output/iteration, so we use a sorted() to always get the same results. A bit of overhead, but again, good enough for now